core/src/test/java/org/elasticsearch/index/query/RangeQueryBuilderTest.java

```diff
@@ -45,32 +45,42 @@
-                // use unmapped field for numeric range queries
-                query = new RangeQueryBuilder(randomAsciiOfLengthBetween(1, 10));
-                query.from(0.0 - randomDouble());
-                query.to(randomDouble());
-            }
-        } else {
-            // use mapped date field, using date string representation
-            query = new RangeQueryBuilder(DATE_FIELD_NAME);
-            query.from(new DateTime(System.currentTimeMillis() - randomIntBetween(0, 1000000), DateTimeZone.UTC).toString());
-            query.to(new DateTime(System.currentTimeMillis() + randomIntBetween(0, 1000000), DateTimeZone.UTC).toString());
-            // Create timestamp option only then we have a date mapper, otherwise we could trigger exception.
-            if (createContext().mapperService().smartNameFieldType(DATE_FIELD_NAME) != null) {
+        switch (randomIntBetween(0, 2)) {
+            case 0:
+                query = createRandomStringRangeQuery();
+                break;
+            case 1:
                 if (randomBoolean()) {
-                    query.timeZone(TIMEZONE_IDS.get(randomIntBetween(0, TIMEZONE_IDS.size() - 1)));
+                    // use mapped integer field for numeric range queries
+                    query = new RangeQueryBuilder(INT_FIELD_NAME);
+                    query.from(randomIntBetween(1, 100));
+                    query.to(randomIntBetween(101, 200));
+                } else {
+                    // use unmapped field for numeric range queries
+                    query = new RangeQueryBuilder(randomAsciiOfLengthBetween(1, 10));
+                    query.from(0.0 - randomDouble());
+                    query.to(randomDouble());
                 }
-                if (randomBoolean()) {
-                    query.format("yyyy-MM-dd'T'HH:mm:ss.SSSZZ");
+                ;
+                break;
+            case 2:
+                // use mapped date field, using date string representation
+                query = new RangeQueryBuilder(DATE_FIELD_NAME);
+                query.from(new DateTime(System.currentTimeMillis() - randomIntBetween(0, 1000000), DateTimeZone.UTC).toString());
+                query.to(new DateTime(System.currentTimeMillis() + randomIntBetween(0, 1000000), DateTimeZone.UTC).toString());
+                // Create timestamp option only then we have a date mapper,
+                // otherwise we could trigger exception.
+                if (createContext().mapperService().smartNameFieldType(DATE_FIELD_NAME) != null) {
+                    if (randomBoolean()) {
+                        query.timeZone(TIMEZONE_IDS.get(randomIntBetween(0, TIMEZONE_IDS.size() - 1)));
+                    }
+                    if (randomBoolean()) {
+                        query.format("yyyy-MM-dd'T'HH:mm:ss.SSSZZ");
+                    }
                 }
-            }
+                break;
+            default:
+                query = createRandomStringRangeQuery();
```

Comment:
Okay, I see why less dependencies between tests is good. This will lead to a bit of code duplication, since I think extending coverage to the String-field case in the RangeQueryTest is a good idea anyway.